card-gallery/script.js

```diff
@@ -263,23 +275,31 @@ function submitSearch() {
           $(this).hide();
         }
       })
-    } catch(e){}
+    } catch (ex) {
+      // If we catch an error with the regex, swallow it
+      console.log("Caught an exception when performing a regex search", ex);
+    }
+    return;
   }
-  else {
-    // Get the query and make it not case-sensitive
-    const query = $(".searchInput").val().toLowerCase();
-
-    // Show or hide each card
-    $(".card").each(function (index, element) {
-      // Get card content (text of cardDetails element basically) and make it not case-sensitive
-      let cardContent = $(this).text().toLowerCase();
-      // See if the card content contains the query
-      if (cardContent.includes(query)) {
-        $(this).show();
-      }
-      else {
-        $(this).hide();
-      }
-    })
+  
+  // Try to perform an expressive search. If that fails (not out of the question), fall back to legacy search.
+  if(expressiveSearch($(".searchInput").val().toLowerCase())) {
+    return;
   }
-}
\ No newline at end of file
+
+  // Get the query and make it not case-sensitive
+  const query = $(".searchInput").val().toLowerCase();
```

Comment:
Just to be clear: I'm not getting rid of the previous way of searching, but I'm making expressive search the first line and then falling back.

Eventually I think we should get rid of "legacy" search, but expressive search needs to support OR queries before it can reach feature parity, and ORing is going to be a whole 'nother bucket of worms.